### PR TITLE
Add login feature

### PR DESCRIPTION
**Pull request by ntotten-kainos**
Add login and registration pages.

Roles are being added at a later date as part of another ticket. This will allow us to differentiate between Applicants, Admins, and Recruiters without having multiple login screens, user classes etc. But finer implementation details remain to be decided.

Login Screen:
image

Register Screen: 
image

### CODE REVIEW

File: src/JobApplicationMVCApp/Models/ApplicationUser.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+using Microsoft.AspNetCore.Identity;
+
+namespace JobApplicationMVCApp.Models
+{
+    public class ApplicationUser : IdentityUser
+    {
+
```
Comment (Conaill4) [inline src/JobApplicationMVCApp/Models/ApplicationUser.cs:7]: is this needed?

File: src/JobApplicationMVCApp/Areas/Identity/Pages/Account/ResetPasswordConfirmation.cshtml.cs
Comment (Conaill4) [inline src/JobApplicationMVCApp/Areas/Identity/Pages/Account/ResetPasswordConfirmation.cshtml.cs:1]: is this also needed?

File: src/JobApplicationMVCApp/Areas/Identity/Pages/Account/Register.cshtml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    <div class="govuk-grid-column-one-third">
+        <section>
+            <h3>Use another service to register.</h3>
+            <hr />
+            @{
+                if ((Model.ExternalLogins?.Count ?? 0) == 0)
+                {
+                    <div>
+                        <p>
+                            There are no external authentication services configured. See this <a href="https://go.microsoft.com/fwlink/?LinkID=532715">article
```
Comment (Conaill4) [inline src/JobApplicationMVCApp/Areas/Identity/Pages/Account/Register.cshtml:42]: maybe hide this from showing on the page

File: src/JobApplicationMVCApp/Areas/Identity/Pages/Account/Login.cshtml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    <div class="govuk-grid-column-one-third">
+        <section>
+            <h3>Use another service to log in.</h3>
+            <hr />
+            @{
+                if ((Model.ExternalLogins?.Count ?? 0) == 0)
+                {
+                    <div>
+                        <p>
+                            There are no external authentication services configured. See this <a href="https://go.microsoft.com/fwlink/?LinkID=532715">article
```
Comment (Conaill4) [inline src/JobApplicationMVCApp/Areas/Identity/Pages/Account/Login.cshtml:61]: hide this too?

File: src/JobApplicationMVCApp/Areas/Identity/Pages/Account/ResetPasswordConfirmation.cshtml.cs
Comment (ntotten-kainos) [reply src/JobApplicationMVCApp/Areas/Identity/Pages/Account/ResetPasswordConfirmation.cshtml.cs:1]: Just part of the scaffolding, figured easier to keep until we formally decide to remove or not and save the hassle of re-scaffolding Identity stuff. But I might be missing something

File: src/JobApplicationMVCApp/Models/ApplicationUser.cs (hunk 1 above)
Comment (ntotten-kainos) [reply src/JobApplicationMVCApp/Models/ApplicationUser.cs:7]: Unsure, if we decide to add custom User fields then yes, if not then no. It's also just an effect of the scaffolding. Creates a default IdentityUser class or uses an implicit IdentityUser with no class declared

**Review by Conaill4 — APPROVED** [review]

**Pull request merged**